conans/cli/output.py

```diff
@@ -0,0 +1,221 @@
+    BRIGHT_MAGENTA = Style.BRIGHT + Fore.MAGENTA  # @UndefinedVariable
+
+
+if get_env("CONAN_COLOR_DARK", 0):
+    Color.WHITE = Fore.BLACK
+    Color.CYAN = Fore.BLUE
+    Color.YELLOW = Fore.MAGENTA
+    Color.BRIGHT_WHITE = Fore.BLACK
+    Color.BRIGHT_CYAN = Fore.BLUE
+    Color.BRIGHT_YELLOW = Fore.MAGENTA
+    Color.BRIGHT_GREEN = Fore.GREEN
+
+try:
+    from logging import NullHandler
+except ImportError:
+    class NullHandler(logging.Handler):
+        def handle(self, record):
+            pass
+
+        def emit(self, record):
+            pass
+
+        def createLock(self):
+            self.lock = None
+
+
+class TqdmHandler(logging.StreamHandler):
+    def __init__(self, stream=None):
+        self._stream = stream
+        super(TqdmHandler, self).__init__(stream)
+
+    def emit(self, record):
+        try:
+            msg = self.format(record)
+            tqdm.tqdm.write(msg, file=self._stream)
+            self.flush()
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except:
+            self.handleError(record)
+
+
+class ConanOutput(object):
+    def __init__(self, stream=None, color=False):
+        self._logger_name = "conan.output"
+        self._stream = stream
+        self._stream_handler = None
+
+        if self._stream is None:
+            logging.getLogger(self._logger_name).addHandler(NullHandler())
+        else:
+            self._stream_handler = TqdmHandler(self._stream)
+            self._stream_handler.setFormatter(logging.Formatter("%(message)s"))
+            logging.getLogger(self._logger_name).addHandler(self._stream_handler)
+            logging.getLogger(self._logger_name).setLevel(logging.INFO)
+
+            logging.captureWarnings(True)
+            logging.getLogger("py.warnings").setLevel(logging.INFO)
+            logging.getLogger("py.warnings").addHandler(self._stream_handler)
+
+        self._color = color
+        self._scope = None
+
+    @property
+    def stream(self):
+        return self._stream
+
+    @property
+    def color(self):
+        return self._color
+
+    @property
+    def scope(self):
+        return self._scope
+
+    @scope.setter
+    def scope(self, out_scope):
+        self._scope = out_scope
+
+    @property
+    def is_terminal(self):
+        return hasattr(self._stream, "isatty") and self._stream.isatty()
+
+    def _write(self, msg, level=logging.INFO, fg=None, bg=None):
+        if self._scope:
+            msg = "{}: {}".format(self.scope, msg)
+        if self._color:
+            msg = "{}{}{}{}".format(fg or '', bg or '', msg, Style.RESET_ALL)
+        for _ in range(3):
+            try:
+                logging.getLogger(self._logger_name).log(level, msg)
+                break
+            except IOError:
+                import time
+                time.sleep(0.02)
+            except UnicodeError:
+                msg = msg.encode("utf8").decode("ascii", "ignore")
+
+        self.flush()
+
+    def debug(self, msg):
```

Comment:
I'm not sure if we are concerned about performance for output, do we want to have here the `*args/**kwargs` so we can avoid string formatting if the logger doesn't have any handler?

(same comment for the other levels)